### DISCUSSION TITLE
Get Data From React Admin Field

### DISCUSSION THREAD

**Discussion by fauh45** [react-page/react-page#1192] [Q&A]
Currently what I see from the react admin integration is that we can get a reference from another React Admin resource. But what if what I need is for the element to use the other fields of the same place, is that possible?

Let's say there's a resource "Page", there's field "name" and then there's the React Page field, how do I make a component that uses the "name" field on it?

**Comment by macrozone** [comment]
@fauh45 i don't understand the question. Can you try to rephrase it, add some examples and ideally some code?

**Reply by fauh45** [reply]
@macrozone 

So maybe an example is, I have this one field in my React Admin edit forms, let's say it's an id of some category. Now how do I use that data from the field, so that when I change that data there it will reflect it into the custom component that I build.

**Reply by macrozone** [reply]
I suggest to first think about how this component work without react-admin and react-page.

I assume you will display the component read only somewhere, so it has to work independently.

Think about the props of the component. If its a component that loads some data through something like graphql, rest, or similar you probably want to just pass the id of a record (or similar) to the component and the component handles the data loading.

if you then create a CellPlugin for react-page that wraps this component, it just needs a form field for the id. Simplest case is just to have an input and the content manager has to type in the id. To make that more convenient, you can create a [custom field](https://react-page.github.io/docs/#/custom-cell-plugins?id=custom-field-components-example)
